app/main.py

```diff
@@ -1,8 +1,24 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[self.name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people: list[dict]) -> list[Person]:
+    person_list = []
+
+    for man in people:
+        person = Person(man["name"], man["age"])
+        person_list.append(person)
+
+    for man in people:
+        if "wife" in man and man["wife"] is not None:
+            husband = Person.people[man["name"]]
```

Comment:
I believe this logic should be a part of the another branch where "husband" is in `man` dict